File: lang/codes/base.c/src/c/inf.c/src/led.c
#include "led.h"
#include "light.h"

int LedOn(struct LedImpl *self);
int LedOff(struct LedImpl *self);
int LedState(struct LedImpl *self);

struct LightInterface LedInf = {
        .on = (LightOnFun)LedOn,
        .off = (LightOffFun)LedOff,
        .state = (LightStateFun)LedState,
};


int LedOn(struct LedImpl *self)
{
    self->state = 1;
    return 0;
}

int LedOff(struct LedImpl *self)
{
    self->state = 0;
    return 0;
}

int LedState(struct LedImpl *self)
{
    return self->state;
}

int NewLedImpl(struct LedImpl *self)
{
    self->inf = &LedInf;
    self->state = 0;
    return 0;
}

File: lang/codes/jdemo/src/main/java/org/example/Led.java
package org.example;

import org.example.inf.Light;

public class Led implements Light {

    public String status;

    public Led() {
        status = "off";
    }

    @Override
    public void on() {
        // TODO
        status = "on";
        System.out.println("Led on ...");
    }

    @Override
    public void off() {
        // TODO
        status = "off";
        System.out.println("Led off ...");
    }

    @Override
    public String getStatus() {
        return status;
    }
}
